apps/dashboard/src/contract-ui/tabs/tokens/components/claim-button.tsx

```diff
@@ -1,37 +1,171 @@
+      <SheetTrigger asChild>
+        <Button
+          colorScheme="primary"
+          leftIcon={<Gem size={16} />}
+          {...restButtonProps}
+        >
+          Claim
+        </Button>
+      </SheetTrigger>
+      <SheetContent className="z-[10000]">
+        <SheetHeader>
+          <SheetTitle>Claim tokens</SheetTitle>
+        </SheetHeader>
+        <form>
+          <div className="flex flex-col gap-6 mt-10">
+            <FormControl isRequired isInvalid={!!form.formState.errors.to}>
+              <FormLabel>To Address</FormLabel>
+              <Input
+                placeholder={ZERO_ADDRESS}
+                {...form.register("to", { required: true })}
+              />
+              <FormHelperText>Enter the address to claim to.</FormHelperText>
+              <FormErrorMessage>
+                {form.formState.errors.to?.message}
+              </FormErrorMessage>
+            </FormControl>
+            <FormControl isRequired isInvalid={!!form.formState.errors.amount}>
+              <FormLabel>Amount</FormLabel>
+              <Input
+                type="text"
+                pattern={`^\\d+(\\.\\d{1,${_decimals || 18}})?$`}
+                {...form.register("amount", { required: true })}
+              />
+              <FormHelperText>How many would you like to claim?</FormHelperText>
+              <FormErrorMessage>
+                {form.formState.errors.amount?.message}
+              </FormErrorMessage>
+            </FormControl>
+          </div>
+        </form>
+        <SheetFooter className="mt-10">
+          <TransactionButton
+            transactionCount={1}
+            form={CLAIM_FORM_ID}
+            isLoading={form.formState.isSubmitting}
+            type="submit"
+            colorScheme="primary"
+            isDisabled={!form.formState.isDirty || isLoading}
+            onClick={form.handleSubmit(async (d) => {
+              if (!d.to) {
+                return toast.error(
+                  "Need to speficy an address to receive tokens",
+                );
+              }
+              trackEvent({
+                category: "token",
+                action: "claim",
+                label: "attempt",
+              });
+              if (!account) {
+                return toast.error("No account detected");
+              }
+              const transaction = ERC20Ext.claimTo({
+                contract,
+                to: d.to,
+                quantity: d.amount,
+                from: account.address,
+              });
+
+              const approveTx = await ERC20Ext.getApprovalForTransaction({
+                transaction,
+                account,
+              });
+
+              if (approveTx) {
+                try {
+                  await sendAndConfirmTransaction.mutateAsync(approveTx);
+                } catch {
+                  return toast.error("Error approving ERC20 token");
+                }
+              }
```

Comment:
you can do the same `toast.promise` here while u wait for this to be confirmed (that way user sees what's going on)
```
const promise = sendAndConfirmTransaction.mutateAsync(approveTx);
toast.promise(promise, {...})
await promise // to actually wait for the promise to finish at runtime
...
```

Reply:
ty ser. I was thinking about how I would want to execute this. Updated the code. Let's see what you think about it ^